src/lib.rs

```diff
@@ -980,17 +980,24 @@ fn process_dom_node<'a, 'b, T: Write>(
                 expanded_name!(html "img") => {
                     let borrowed = attrs.borrow();
                     let mut title = None;
+                    let mut src = None;
                     for attr in borrowed.iter() {
                         if &attr.name.local == "alt" && !attr.value.is_empty() {
                             title = Some(&*attr.value);
+                        }
+                        if &attr.name.local == "src" && !attr.value.is_empty() {
+                            src = Some(&*attr.value);
+                        }
+                        if title.is_some() && src.is_some() {
                             break;
                         }
                     }
-                    if let Some(title) = title {
-                        Finished(RenderNode::new(Img(title.into())))
-                    } else {
-                        Nothing
-                    }
+                    return match (title, src) {
```

Comment:
This shouldn't be a return statement -  just the `match` without terminating semicolon.
It could still be `if let`, but doesn't matter too much either way:
```rust
if let ((Some(title), Some(src)) = (title, src) {
    ...
} else {
    Nothing
}
```

Reply:
Yeah, I think that structure is clearer. Updated to use an `if let`